src/plugins/intel_cpu/src/nodes/memory.cpp

```diff
@@ -118,57 +235,189 @@ MemoryInput::MemoryInput(const std::shared_ptr<ov::Node>& op, const GraphContext
 
 void MemoryInput::createPrimitive() {
     Input::createPrimitive();
+    if (!inputShapes.empty()) {
+        auto parentEdge = getParentEdgeAt(0);
 
-    dataStore = std::make_shared<Memory>(getEngine(), getChildEdgeAt(0)->getMemory().getDesc());
+        if (parentEdge->getParent()->isConstant()) {
+            Input::resetMemoryPtr(parentEdge->getMemoryPtr());
+        }
+    }
+}
 
-    // default memory state is zero filled
-    if (dataStore->getDesc().hasDefinedMaxSize())
-        dataStore->nullify();
+void MemoryInput::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    auto&& shape = getOutputShapeAtPort(0);
+    auto precision = getOriginalOutputPrecisionAtPort(0);
+    auto&& descCreators = ov::intel_cpu::BlockedDescCreator::getCommonCreators();
+
+    NodeConfig config;
+
+    if (!getParentEdges().empty()) {
+        PortConfig inPortConfig;
+
+        inPortConfig.inPlace(-1);
+        inPortConfig.constant(false);
+        inPortConfig.setMemDesc(descCreators.at(LayoutType::ncsp)->createSharedDesc(precision, shape));
+
+        config.inConfs.push_back(std::move(inPortConfig));
+    }
+
+    PortConfig outPortConfig;
+
+    outPortConfig.inPlace(0);
```

Comment:
I think here we simply reuses the interface of inPlace, but changes its original meaning, right?
outPortConfig.inPlace(0) originally means this outport reuses memory from input port 0. But obviously, it is not. The memory is from state variable instead.

Reply:
That's why you `resolve inplace dead end nodes` in the beginning of `Graph::Allocate()`, right?

Reply:
Yep, exactly. This memory substitution fits into the inplace memory model completely, except the constraints on its description. So there was to options: 1. Add a fake edge with the memory from the state 2. Remove constraints on persistence of the edge that is referenced. First option is more bulky as requires a special treatment from the graph logic, while the second one requires only one additional condition for the dead end nodes.